lib/landsat_calibration.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Tue Dec 15 12:52:27 2015
+
+@author: Janek Mazur, Hubert Zieliński, Paweł Przychodzień
+"""
+import gdal, gdalconst, numpy, time
+
+def get_input_dat(input_path):
+    '''Otwiera obraz Landsata 8. Jako parametr wejściowy przyjmuje scieżkę
+    do pliku .bsq'''
+    sterownik=gdal.GetDriverByName('ENVI')
+    sterownik.Register()
+    input_dat=gdal.Open(input_path,gdalconst.GA_ReadOnly)
+    return input_dat
+    
+def upload_band(input_dat,bandX):
+    '''Pobiera wartości zadanego kanału z określonego zobrazowania landsat 8.
+    Jako parametry wejściowe przyjmuje cieżkę do pliku obrazu *.bsq i numer kanału
+    ktory ma zostac zaimportowany. Wartosci kanału sa dodawane do macieży.'''
+    kanal=input_dat.GetRasterBand(bandX)
+    matrix=numpy.array(kanal.ReadAsArray(),dtype=numpy.float32)
```

Comment:
Nie musimy tu rzutować. Ale wg. mnie nie ma to znaczenia bo jak ustaliliśmy na zajęciach gdal nie obsługuje float 16, zatem i tak będziemy musieli rzutować to na float 32. Funkcja upload_band jest wywoływana przez funkcje band _reflectance w funkcji zapisu, czyli tuż przed momentem w którym dane i tak muszą zostać rzutowane do float 32.